src/microbe_stage/Microbe.cs

```diff
@@ -904,10 +904,34 @@ public void AIThink(float delta, Random random, MicrobeAICommonData data)
     /// </returns>
     public List<(Compound Compound, Color Colour, Vector3 Target)> GetDetectedCompounds(CompoundCloudSystem clouds)
     {
+        HashSet<(Compound Compound, float Range, float MinAmount, Color Colour)> colonyCompoundDetections = new();
```

Comment:
This still allocates an unnecessary object for the non-colony case.

Reply:
I moved it.